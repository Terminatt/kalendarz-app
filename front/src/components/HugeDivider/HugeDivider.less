@import '~styles/variables.less';

.ant-divider-horizontal.ant-divider-with-text.huge-divider {
  border-top-color: @primary-color;

  &::before, &::after {
    border-width: 3px;
  }

  & > .ant-divider-inner-text {
    font-size: @font-size-enormous;
    padding: 0 @padding-big;
  }
}