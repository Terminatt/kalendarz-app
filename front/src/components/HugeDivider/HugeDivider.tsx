import { Divider, DividerProps } from 'antd';
import React from 'react';

import './HugeDivider.less';

export interface HugeDividerProps extends DividerProps {
  text: string | number;
}

const HugeDivider: React.FC<HugeDividerProps> = (props) => {
    const { text, className, ...rest } = props;

    return (
        <Divider className={`huge-divider ${className || ''}`} {...rest}>
            {text}
        </Divider>
    );
};

export default HugeDivider;
