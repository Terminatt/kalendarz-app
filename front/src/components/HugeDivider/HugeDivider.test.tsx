import { render } from '@testing-library/react';
import React from 'react';
import renderer from 'react-test-renderer';
import HugeDivider from './HugeDivider';

describe('Huge Divider Component', () => {
    it('matches the snapshot', () => {
        const tree = renderer
            .create(
                <HugeDivider text="Test" />
            )
            .toJSON();
        
        expect(tree).toMatchSnapshot();
    });

    it('renders text', () => {
        const text = 'Test'
        const screen = render(
            <HugeDivider text={text} />
        );
        expect(screen.getByText(text)).not.toBeNull();
    });
});
