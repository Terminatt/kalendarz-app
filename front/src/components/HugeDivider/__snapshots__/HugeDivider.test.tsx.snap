// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Huge Divider Component matches the snapshot 1`] = `
<div
  className="ant-divider ant-divider-horizontal ant-divider-with-text ant-divider-with-text-center huge-divider"
  role="separator"
>
  <span
    className="ant-divider-inner-text"
  >
    Test
  </span>
</div>
`;
